File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.py
from collections import deque

def solution(cards1, cards2, goal):
    queue1 = deque(cards1)
    queue2 = deque(cards2)
    goal_queue = deque(goal)
    while len(goal_queue) > 0:
        word = goal_queue.popleft()
        if queue1 and queue1[0] == word:
            queue1.popleft()
            continue
        if queue2 and queue2[0] == word:
            queue2.popleft()
            continue
        return "No"
    return "Yes"

File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.js
function solution(cards1, cards2, goal) {
    for (i in goal) {
        const word = goal[i];
        if (cards1[0] === word) {
            cards1.shift();
            continue;
        }    
        if (cards2[0] === word) {
            cards2.shift();
            continue;
        }
        return "No";
    }
    return "Yes";
}